infra/src/main/java/com/redhat/service/bridge/infra/models/filters/FilterFactory.java

```diff
@@ -0,0 +1,16 @@
+package com.redhat.service.bridge.infra.models.filters;
+
+public class FilterFactory {
+    public static Filter buildFilter(String type, String key, String value) {
+        switch (type) {
+            case StringBeginsWith.FILTER_TYPE_NAME:
+                return new StringBeginsWith(key, value);
+            case StringContains.FILTER_TYPE_NAME:
+                return new StringContains(key, value);
+            case StringEquals.FILTER_TYPE_NAME:
+                return new StringEquals(key, value);
+            default:
+                throw new IllegalArgumentException("Type " + type + " is not valid");
+        }
+    }
```

Comment:
The manager stores the `value` of a Filter as String in the database, then it needs to convert back to a Filter using this factory

Reply:
Just for visibility, JPA supports `@PostLoad` and `@PostUpdate` callbacks that might be another option for this: https://nullbeans.com/configuring-postload-and-postupdate-in-jpa-hibernate/

I think this is fine for now though as it's clear as to what's going on, but we might want to consider if it makes our life easier in the future.

Reply:
Created https://issues.redhat.com/browse/MGDOBR-42